sklearn/preprocessing/imputation.py

```diff
@@ -143,27 +147,35 @@ def fit(self, X, y=None):
                              " got strategy={1}".format(allowed_strategies,
                                                         self.strategy))
 
-        if self.axis not in [0, 1]:
+        if self.axis is None:
+            self._axis = 0
+        else:
+            warnings.warn("'axis' will be removed from Imputer, and it will "
```

Comment:
The convention is to add the version the deprecation happened as well as the on it will be removed.

From http://scikit-learn.org/stable/developers/contributing.html#deprecation:
> As in these examples, the warning message should always give both the version in which the deprecation happened and the version in which the old behavior will be removed.